expr/functions/round/function.go

```diff
@@ -56,19 +56,14 @@ func (f *round) Do(ctx context.Context, e parser.Expr, from, until int64, values
 		r.Values = make([]float64, len(a.Values))
 
 		for i, v := range a.Values {
-			r.Values[i] = doRound(v, precision)
+			r.Values[i] = helper.SafeRound(v, precision)
 		}
-		results[j] = &r
-	}
-	return results, nil
-}
 
-func doRound(x float64, precision int) float64 {
-	if math.IsNaN(x) {
-		return x
+		r.Tags["round"] = fmt.Sprintf("%d", precision)
```

Comment:
We really need string conversion in each call of function ? We have preconverted precisionStr.